Sources/Kafka/Configuration/KafkaConfiguration+Security.swift

```diff
@@ -122,104 +118,123 @@ extension KafkaConfiguration {
+                return .init(_internal: .keyStore(keyStore: keyStore))
+            }
         }
 
-        let _internal: _TLSConfiguration
-
-        /// Use TLS with a given private/public key pair.
-        ///
-        /// - Parameters:
-        ///
-        ///     - privateKey: The client's private key (PEM) used for authentication.
-        ///     - publicKeyCertificate: The client's public key (PEM) used for authentication.
-        ///     - caCertificate: File or directory path to CA certificate(s) for verifying the broker's key.
-        ///     - crLocation: Path to CRL for verifying broker's certificate validity.
-        public static func keyPair(
-            privateKey: PrivateKey,
-            publicKeyCertificate: LeafAndIntermediates,
-            caCertificate: Root = .probe,
-            crlLocation: String?
-        ) -> TLSConfiguration {
-            return TLSConfiguration(
-                _internal: .keyPair(
-                    privateKey: privateKey,
-                    publicKeyCertificate: publicKeyCertificate,
-                    caCertificate: caCertificate,
-                    crlLocation: crlLocation
+        /// Configuration for the TLS verification of the broker.
+        public struct BrokerVerification: Sendable, Hashable {
+            internal enum _BrokerVerification: Sendable, Hashable {
+                case disable
+                case verify(
+                    rootCertificates: TrustRoots,
+                    crlLocation: String?
                 )
-            )
-        }
+            }
+
+            let _internal: _BrokerVerification
+
+            /// Do not verify the identity of the broker.
+            public static let disable: BrokerVerification = .init(_internal: .disable)
 
-        ///
-        /// - Parameters:
-        ///
-        ///     - keyStore: The client's keystore (PKCS#12) used for authentication.
-        ///     - caCertificate: File or directory path to CA certificate(s) for verifying the broker's key.
-        ///     - crlLocation: Path to CRL for verifying broker's certificate validity.
-        public static func keyStore(
-            keyStore: KeyStore,
-            caCertificate: Root = .probe,
-            crlLocation: String?
-        ) -> TLSConfiguration {
-            return TLSConfiguration(
-                _internal: .keyStore(
-                    keyStore: keyStore,
-                    caCertificate: caCertificate,
-                    crlLocation: crlLocation
+            /// Verify the identity of the broker.
+            ///
+            /// Parameters:
+            ///     - caCertificate: File or directory path to CA certificate(s) for verifying the broker's key.
+            ///     - crlLocation: Path to CRL for verifying broker's certificate validity.///
+            public static func verify(
+                rootCertificates: TrustRoots = .probe,
+                crlLocation: String?
```

Comment:
```suggestion
                trustRoots: TrustRoots = .probe,
                certificateRevocationListPath: String?
```

Reply:
Let's discuss if we should also default the crl to `nil`. I expect not a lot of people will pass a custom one here.